modules/tensor_mechanics/src/actions/DomainIntegralAction.C

```diff
@@ -304,8 +302,11 @@ DomainIntegralAction::act()
     if (_use_crack_front_points_provider)
     {
       params.set<UserObjectName>("crack_front_points_provider") = _crack_front_points_provider;
-      params.set<unsigned int>("number_points_from_provider") = _number_crack_front_points;
+      params.set<unsigned int>("number_points_from_provider") =
+          getParam<unsigned int>("number_points_from_provider");
     }
+    if (_closed_loop)
```

Comment:
You just introduced another error: Up above you conditionally initialized `_closed_loop`, that means that this statement works half the time and returns garbage the other half of the time. 

I don't immediately see where the parameter closed_loop is coming from, perhaps it's already in the code base and is being inherited? It wasn't added in this PR. Can you just provide a default value for that parameter and **always** initialize the class member? Less code and bug free ;)

Reply:
Good point. Thanks!